Assets/Scripts/Interactables/Dialogue/NPCDialogue/NPCInteractable.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[System.Serializable]
+public class MissionNPC
+{
+    public GameObject missionPrefab;
+
+    // Type of conversations an NPC can have based on if it is before, during or after a
+    public Conversation beforeConvo;
+    public Conversation duringConvo;
+    public Conversation afterConvo;
+}
+
+public class NPCInteractable : DialogueInteractable
+{
+    public List<Conversation> defaultConvos;
+
+    public List<MissionNPC> missionsOffered;
+    private MissionNPC currentMissionNPC;                            // Current Mission given by this NPC (should only be 1 per NPC)
+    private AMission startedMission;                                 // started mission needed to end mission
+
+
+    public override void OnInteract()
+    {
+        if(!isConversing)
+        {
+            // Load the conversation of the NPC based on mission progress
+            if (missionsOffered.Count == 0)
+            {
+                // Return a random default convo (Temporary until better system is set for default convos)
+                conversation = defaultConvos[Random.Range(0, defaultConvos.Count - 1)];
```

Comment:
I think should just be `defaultConvos.Count` without the `-1`? 
`Random.Range` is like `[start, finish)` ya know
id test it out to make sure though

Reply:
Ok yeah it min inclusive and max exclusive!